src/controllers/driverController.js

```diff
@@ -0,0 +1,270 @@
+import bcrypt from "bcrypt";
+import "dotenv/config";
+import getPassword from "../services/createPassword.js";
+import sendEmail from "../services/sendEmail.js";
+import Users from "../../sequelize/models/user.js";
+import Profiles from "../../sequelize/models/profile.js";
+import { Sequelize } from "sequelize";
+import jwt from "jsonwebtoken";
+import { development } from "../../sequelize/config/config.js";
+let sequelize = new Sequelize(development);
+let User = Users(sequelize, Sequelize);
+let Profile = Profiles(sequelize, Sequelize);
+
+class UserController {
+  async registerDatabase(req, res) {
+    const driverRegister = await User.sync({ alter: true });
+    const profileRegister = await Profile.sync({ alter: true });
+    res.send("database created");
+  }
+  async login(req, res) {
+    let isPasswordMatch = "";
+    const project = await User.findOne({ where: { email: req.body.email } });
+    if (project) {
+      isPasswordMatch = await bcrypt.compare(
+        req.body.password,
+        project.password
+      );
+    }
+    if (project === null || !isPasswordMatch) {
+      res.status(404).send({ message: "Incorrect email or password" });
+    } else {
+      const token = jwt.sign(
+        { _id: project.email, role: project.role },
+        process.env.SECRET_KEY
+      );
+      res.status(200).json({
+        token,
+        message: "Login successfully",
+        user: {
+          firstName: project.firstName,
+          lastName: project.lastName,
+          email: project.email,
+          role: project.role,
+        },
+      });
+    }
+  }
+
+  async registerDriver(req, res) {
+    const {
+      firstName,
+      lastName,
+      email,
+      phoneNumber,
+      idNumber,
+      driverLicence,
+      dateofBirth,
+      gender,
+      address,
+    } = req.body;
+    const userpassword = getPassword();
+    const password = await bcrypt.hash(userpassword, 12);
+    const dbDriver = await User.findOne({ where: { email } });
+    let errors = [];
+    // Validate Fields
+    if (dbDriver) {
+      errors.push({ message: "User with email already exist." });
+    }
+    if (!firstName) {
+      errors.push({ message: "Please add a first name" });
+    }
+    if (!dateofBirth) {
+      errors.push({ message: "Please add your date of birth" });
+    }
+    if (!gender) {
+      errors.push({ message: "Please add your gender" });
+    }
+    if (!lastName) {
+      errors.push({ message: "Please add a last name" });
+    }
+    if (!email) {
+      errors.push({ message: "Please add a email" });
+    }
+    if (!phoneNumber) {
+      errors.push({ message: "Please add a phoneNumber" });
+    }
+    if (!idNumber) {
+      errors.push({ message: "Please add a id number" });
+    }
+    if (!driverLicence) {
+      errors.push({ message: "Please add a driver Licence" });
+    }
+    if (!address) {
+      errors.push({ message: "Please add a address" });
+    }
+
+    if (errors.length > 0) {
+      res.status(401).json({
```

Comment:
The Http code for a bad request is 400, not 401. Do some reseach about the error codes.
The http code for an already existing user id 409, conflict.
This generic way of error handling that you used here does not allow you to return the right error error codes, please change that